submodule.go

```diff
@@ -267,7 +267,7 @@ func (s *Submodule) fetchAndCheckout(
 				RefSpecs: []config.RefSpec{refSpec},
 				Depth:    o.Depth,
 			})
-			if err != nil && err != NoErrAlreadyUpToDate && err != ErrExactSHA1NotSupported {
+			if err != nil && errors.Is(err, NoErrAlreadyUpToDate) && errors.Is(err, ErrExactSHA1NotSupported) {
```

Comment:
Not sure I am missing something, isn't this check the opposite of the previous code?
```suggestion
			if err != nil && !errors.Is(err, NoErrAlreadyUpToDate) && !errors.Is(err, ErrExactSHA1NotSupported) {
```

Reply:
Oof, good catch @pjbgf!